solutions/ac-GCDOnBlackboard.mdx

```diff
@@ -76,4 +76,57 @@ int main(){
 
 ```
 </CPPSection>
+	
+<JavaSection>
+
+```java
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class Main {
+	public static void main(String[] args) throws IOException {
+		// Open System.in and read data
+		BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
+		int N = Integer.parseInt(input.readLine()); // N is the length of the array
+		int[] nums = new int[N]; // The nums array (read from input)
+		String[] stringNums = input.readLine().split(" "); // The string representation of nums
+		
+		// Convert string nums into integers
+		for (int i = 0; i < N; i++) {
+			nums[i] = Integer.parseInt(stringNums[i]);
+		}
+		input.close(); // Close input
+
+		// Prefix GCD array, and suffixGCD array
+		int[] prefixGCD = new int[N + 1], suffixGCD = new int[N + 1];
+
+		// Calculate prefix GCD (build from last prefixGCD)
+		for (int i = 1; i <= N; i++) {
+			prefixGCD[i] = gcd(prefixGCD[i - 1], nums[i - 1]);
+		}
+
+		// Calculate suffix GCD (build from forward suffixGCD)
+		for (int i = N - 1; i >= 0; i--) {
+			suffixGCD[i] = gcd(suffixGCD[i + 1], nums[i]);
+		}
+
+		// Find the maximum GCD
+		int maxGCD = 0;
+		for (int i = 1; i <= N; i++) {
+			maxGCD = Math.max(maxGCD, gcd(prefixGCD[i - 1], suffixGCD[i]));
+		}
+
+		// Output
```

Comment:
```suggestion
```
comments need not state the obvious
so uh i really appreciate the effort, but all of the comments (except the one on line 124) are unnecessary
instead i was hoping for some comments that explain like "prefix gcd contains the gcd of every prefix of the array" and something analogous for suffixGCD